File: php/laravel/app/Models/Transfer.php
<?php

namespace App\Models;

use App\Enums\TransferType;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Transfer extends Model
{

    public $timestamps = ['created_at'];
    const UPDATED_AT = null;

    protected $fillable = [
        'sender_user_id',
        'receiver_user_id',
        'sender_account_id',
        'receiver_account_id',
        'type', 'value',
    ];

    protected $casts = [
        'type' => TransferType::class
    ];

    public function sender_user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'sender_user_id');
    }

    public function receiver_user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'receiver_user_id');
    }

    public function sender_account(): BelongsTo
    {
        return $this->belongsTo(BankAccount::class, 'sender_account_id');
    }

    public function receiver_account(): BelongsTo
    {
        return $this->belongsTo(BankAccount::class, 'receiver_account_id');
    }
}

File: api/go/gin/app/models/transfer.go
package models

import (
	"time"
)

type Transfer struct {
	ID                uint64    `json:"id"`
	SenderUserId      uint64    `json:"sender_user_id"`
	ReceiverUserId    uint64    `json:"receiver_user_id"`
	SenderAccountId   uint64    `json:"sender_account_id"`
	ReceiverAccountId uint64    `json:"receiver_account_id"`
	Type              string    `json:"type"`
	Value             uint      `json:"value"`
	CreatedAt         time.Time `json:"created_at"`

	SenderUser      User        `json:"sender_user" gorm:"foreignKey:sender_user_id"`
	ReceiverUser    User        `json:"receiver_user" gorm:"foreignKey:receiver_user_id"`
	SenderAccount   BankAccount `json:"sender_account" gorm:"foreignKey:sender_account_id"`
	ReceiverAccount BankAccount `json:"receiver_account" gorm:"foreignKey:receiver_account_id"`
}
